docs/blog/2024-09-12-next-gen-module-federation-deployments.md

```diff
@@ -0,0 +1,549 @@
+demonstrate deployment with Zephyr Cloud.
+
+{% callout type="note" title="Zephyr Cloud Nx Recipe" %}
+
+Zephyr Cloud also has a recipe that you can follow to set up Nx React Module Federation with Zephyr
+Cloud [here](https://docs.zephyr-cloud.io/recipes/react-rspack-nx).
+
+{% /callout %}
+
+I'll break this section into sub-steps to make it easier to follow.
+
+### Step 1: Add `package.json` and First Commit
+
+First, we need to add `package.json` files to our `shell` and `remote1` projects. This is used by Zephyr Cloud to
+determine the name and version of the projects for deployment.
+
+Create the following files:
+
+```json {% fileName="shell/package.json" %}
+{
+  "name": "shell",
+  "version": "0.0.0"
+}
+```
+
+```json {% fileName="remote1/package.json" %}
+{
+  "name": "remote1",
+  "version": "0.0.0"
+}
+```
+
+Once this is done, add all staged files and create a git commit.
+
+```shell
+git add .
+git commit -m "initial commit"
+```
+
+### Step 2: Create a GitHub Repository
+
+Next, we need to add a GitHub repository. It can be public or private, it will not affect the deployments by Zephyr
+Cloud.
+
+{% callout type="note" title="Why is this needed?" %}
+Behind the scene, Zephyr Cloud will map your git configuration (remote origin url, organization or username, repository
+name and branch) to deploy your application. Without this step the deployment will fail.
+{% /callout %}
+
+If you don't already have the [GitHub CLI](https://cli.github.com/) installed, I would highly recommend it.
+
+You can then run the following to create your GitHub Repository from your terminal:
+
+```{% commmand="gh repo create" path="~/myorg" %}
+? What would you like to do? Push an existing local repository to GitHub
+? Path to local repository .
+? Repository name nx-zephyr-cloud-example
+? Repository owner Coly010
+? Description Example of Nx React Rspack Module Federation with Zephyr Cloud
+? Visibility Private
+✓ Created repository Coly010/nx-zephyr-cloud-example on GitHub
+  https://github.com/Coly010/nx-zephyr-cloud-example
+? Add a remote? Yes
+? What should the new remote be called? origin
+✓ Added remote https://github.com/Coly010/nx-zephyr-cloud-example.git
+? Would you like to push commits from the current branch to "origin"? Yes
+```
+
+### Step 3: Add Zephyr Cloud
+
+To add Zephyr Cloud itself, we need to install their `zephyr-webpack-plugin`.
+
+{% callout type="note" title="Rspack and Webpack Interop" %}
+The plugin is indeed written for Webpack, however, one of the big advantages of Rspack is the interoperability it offers
+for the Webpack Ecosystem.
+You can learn more about this compatability [here](https://rspack.dev/guide/compatibility/plugin).
+{% /callout %}
+
+Run the following to install the plugin:
+
+```shell
+npm install zephyr-webpack-plugin
+```
+
+Now that the plugin has been installed in our workspace, we need to update the `rspack.config.ts` files to use it.
+This is a very simple two line change:
+
+```ts
```

Comment:
if you use the `{% fileName="..." }` then that'll be displayed at the top of the code fence to make it easier to spot where the change needs to happen

Reply:
Yep, I use this afterwards in when showing the actual example

Reply:
Oh wait, I don't I thought I had